web/beacon-app/src/features/projects/components/NewProject/NewProjectModal.tsx

```diff
@@ -0,0 +1,70 @@
+import { t, Trans } from '@lingui/macro';
+import { Modal } from '@rotational/beacon-core';
+import { useEffect } from 'react';
+import { toast } from 'react-hot-toast';
+
+import { useCreateProject } from '@/features/projects/hooks/useCreateProject';
+import { useFetchTenants } from '@/features/tenants/hooks/useFetchTenants';
+
+import NewProjectForm from './NewProjectForm';
+
+type NewProjectModalProps = {
+  isOpened: boolean;
+  onClose: () => void;
+};
+
+function NewProjectModal({ onClose, isOpened }: NewProjectModalProps) {
+  const { createNewProject, isCreatingProject, wasProjectCreated, hasProjectFailed, error, reset } =
+    useCreateProject();
+  const { tenants } = useFetchTenants();
+  const tenantID = tenants?.tenants[0]?.id;
+  const handleSubmit = async (values: any) => {
+    const payload = {
+      ...values,
+      tenantID: tenantID,
+    };
+    await createNewProject(payload);
+  };
+
+  useEffect(() => {
+    if (wasProjectCreated) {
+      toast.success(t`Success! You have created a new project.`);
+      onClose();
+      reset();
+    }
+  }, [wasProjectCreated, onClose, reset]);
+
+  useEffect(() => {
+    if (hasProjectFailed) {
+      toast.error(
+        (error as any)?.response?.data?.error ||
+          t`Could not create project. Please try again or contact support, if the problem continues.`
+      );
+      reset();
+    }
+  }, [hasProjectFailed, error, reset]);
+
+  return (
+    <>
+      <Modal
+        open={isOpened}
+        onClose={onClose}
+        containerClassName="max-h-[90vh] overflow-scroll max-w-[80vw] lg:max-w-[40vw] no-scrollbar"
+        title={t`Create Project`}
+        data-testid="newProjectModal"
```

Comment:
Nitpick: This `testid` is in camel case while a previous comment mentioned different casing styles for `data-cy`. Should we be consistent or use one case for `data-testid` and a different one for `data-cy`?